### PR TITLE
FUSETOOLS-2550 - UI Tests - Fix broken usage of Preference Pages

### PR DESCRIPTION
**Pull request by tsedmik**
https://issues.jboss.org/browse/FUSETOOLS-2550

Signed-off-by: Tomáš Sedmík <tsedmik@redhat.com>

### CODE REVIEW

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/utils/FuseServerManipulator.java [hunk 1]
```diff
@@ -299,8 +299,8 @@ public static void removeAllModules(String name) {
 		}
 
 		// There is something to remove - remove it
-		new DefaultShell("Add and Remove...");
-		FuseModifyModulesPage page = new FuseModifyModulesPage();
+		DefaultShell shell = new DefaultShell("Add and Remove...");
```
Comment (apodhrad) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/utils/FuseServerManipulator.java]: What about using the following code?
```
List<ServerModule> modules = server.getModules();
for(ServerModule module: modules) {
    module.remove();
}
```

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/utils/FuseServerManipulator.java [hunk 2]
```diff
@@ -358,9 +358,11 @@ public static void addModule(String server, String project) {
...
 	public static void setImmeadiatelyPublishing(String name, boolean value) {
 
-		new ServersView2().getServer(name).addAndRemoveModules();
-		new DefaultShell("Add and Remove...");
-		FuseModifyModulesPage page = new FuseModifyModulesPage();
+		ServersView2 view = new ServersView2();
+		view.open();
+		view.getServer(name).addAndRemoveModules();
+		DefaultShell shell = new DefaultShell("Add and Remove...");
+		FuseModifyModulesPage page = new FuseModifyModulesPage(shell);
```
Comment (apodhrad) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/utils/FuseServerManipulator.java]: method addAndRemoveModules() returns ModifyModulesDialog which can be passed to the constructor FuseModifyModulesPage

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/wizard/FuseModifyModulesPage.java [hunk 3]
```diff
@@ -46,6 +42,6 @@ public void setImmeadiatelyPublishing(boolean value) {
 	 */
 	public void close() {
```
Comment (apodhrad) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/wizard/FuseModifyModulesPage.java]: Why do we have method close() which clicks on 'Finish' button?

**Comment by tsedmik** [comment]
@apodhrad thank you for your review. I made some changes and I hope you'll like them ;-).

**Comment by apupier** [comment]
@tsedmik you need to rebase on master to have a successful PR job build

**Pull request merged**